src/com/google/javascript/jscomp/ProcessCommonJSModules.java

```diff
@@ -1078,6 +1091,11 @@ private void updateNameReference(
             }
           } else if (newNameDeclaration != null) {
             // Variable is already defined. Convert this to an assignment.
+            if (!nameRef.hasChildren()) {
+              parent.detachFromParent();
```

Comment:
Please add a comment explaining why we detach the node and return early at this point.